### DISCUSSION TITLE
How to apply migration to DB ?

### DISCUSSION THREAD

**Discussion by senyg1** [jasontaylordev/CleanArchitecture#612] [General]
I'm trying to use the template and I was able to add migration.
But when trying to apply this migration, i'm getting this error:
"Unable to create an object of type 'ApplicationDbContext'. For the different patterns supported at design time, see https://go.microsoft.com/fwlink/?linkid=851728"
I have just typed "update-database"

**Comment by bramburn** [comment]
This is my cheat sheet i have in the readme.
hope this helps:

```shell
dotnet ef migrations add "SampleMigration" --project src\Infrastructure --startup-project src\WebUI --output-dir Persistence\Migrations
```
run database update
```shell

dotnet ef database update --project src\Infrastructure --startup-project src\WebUI
```
create migration
```shell
dotnet ef migrations add CreateIdentitySchema --project src\Infrastructure --startup-project src\WebUI --output-dir Persistence\Migrations
```

**Reply by senyg1** [reply]
Thanks
I have this error when I try to progress.
The entity type 'DomainEvent' requires a primary key to be defined. If you intended to use a keyless entity type, call 'HasNoKey' in 'OnModelCreating'. For more information on keyless entity types, see https://go.microsoft.com/fwlink/?linkid=2141943.
Have you even met this?

**Reply by bramburn** [reply]
did you create any migrations ontop of the default?

also you might need to disable using the inmemory db configuration in appsettings
